### PR TITLE
enhance: Rootcoord's stop may block in quota_center's stop

### PR DESCRIPTION
**Pull request by weiliu1031**
this PR fixed that rootcoord's stop may block in quota_center' stop

### CODE REVIEW

**Review by congqixia — COMMENTED** [review]
what if stop arrives when `run` goroutine already stuck in syncMetrics or other steps?

**Comment by weiliu1031** [comment]
> what if stop arrives when `run` goroutine already stuck in syncMetrics or other steps?

`syncMetrics`  use ctx with timeout 10s, which means syncMetrics only block 10s in each round

File: internal/rootcoord/quota_center.go [hunk 1]
```diff
@@ -145,6 +145,13 @@ func (q *QuotaCenter) run() {
 			log.Info("QuotaCenter exit")
 			return
 		case <-ticker.C:
+			// syncMetrics may cost several seconds when qc/dc is offline, and with golang's random select policy
+			// quotaCenter' Stop may block here for minutes. which also affect rootcood's stop
+			_, closed := <-q.stopChan
```
Comment (congqixia) [inline internal/rootcoord/quota_center.go:150]: Please see line L177, stopChan is not closed but have a singal
So Signal will never to `true` here

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**